### DISCUSSION TITLE
Confusing terminology: MSB & LSB vs Offset & Width

### DISCUSSION THREAD

**Discussion by iDoka** [esynr3z/corsair#134] [Questions] [answered]
In config used "mixed" terminology: LBS + Width to specify fields size and position inside word.

this is contrary to conventional terminology when used couple of terms: `MSB & LSB` or `Offset & Width`.

As an example of this point: https://github.com/sifive/duh/blob/master/docs/component.md#registers
```
name: 'data', bitOffset: 0, bitWidth: 5
```

It would be nice to add the complementary and meaningful term `offset` to the term `width` (instead currently using `lsb`) 🤓

**Comment by esynr3z** [comment, marked as answer]
Okay it make sense. I will include it into 2.0.

Upd. In v2.x `offset` and `width` are the ones used from user side for a field. `lsb` and `msb` are derived from them and can be used in templates.
